pkg/wallet/wallet.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// BaseWallet wallet implementation
+type BaseWallet struct {
+	store storage.Store
+}
+
+// key json structure include public,private and metadata
+type key struct {
+	Public   string            `json:"public"`
```

Comment:
Is there a reason why `Public` isn't a sequence of bytes as well?

Reply:
base58 encoded

Reply:
it should be same type as Private and convert from string to *[32]byte if needed